python-mini-projects/projects/Scraping Medium Articles/scraping_medium.py

```diff
@@ -0,0 +1,71 @@
+import os
+import sys
+import requests
+import re
+from bs4 import BeautifulSoup
+
+# switching to current running python files directory
+os.chdir('\\'.join(__file__.split('/')[:-1]))
+
+# function to get the html of the page
+def get_page():
+	global url
+	url = input('Enter url of a medium article: ')
+	# handling possible error
+	if not re.match(r'https?://medium.com/',url):
+		print('Please enter a valid website, or make sure it is a medium article')
+		sys.exit(1)
+	res = requests.get(url)
+	res.raise_for_status()
+	soup = BeautifulSoup(res.text, 'html.parser')
+	return soup
+
+# function to remove all the html tags and replace some with specific strings
+def purify(text):
+    rep = {"<br>": "\n", "<br/>": "\n", "<li>":  "\n"}
+    rep = dict((re.escape(k), v) for k, v in rep.items()) 
+    pattern = re.compile("|".join(rep.keys()))
+    text = pattern.sub(lambda m: rep[re.escape(m.group(0))], text)
+    text = re.sub('\<(.*?)\>', '', text)
+    return text
+
+# function to compile all of the scraped text in one string
+def collect_text(soup):
+	fin = f'url: {url}\n\n'
+	main = (soup.head.title.text).split('|')
+	global title
+	title = main[0].strip()
+	fin += f'Title: {title.upper()}\n{main[1].strip()}'
+
+	header = soup.find_all('h1')
+	j = 1
+
+	try:
+		fin += '\n\nINTRODUCTION\n'
+		for elem in list(header[j].previous_siblings)[::-1]:
+			fin += f'\n{purify(str(elem))}'
+	except:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are generically passing an `Exception` object without performing any other operation on it. This may hide error conditions that can otherwise be quickly detected and addressed. We recommend that you catch a more specific exception. If the code must broadly catch all exceptions, consider logging the stack trace using the [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) API. For example,

```
try:
    x = 1 / 0
except ZeroDivisionError as e:
    logging.exception(&#39;ZeroDivisionError: %s&#39;, e)

```